custom_components/opnsense/helpers.py

```
@@ -24,6 +34,80 @@ def dict_get(data: MutableMapping[str, Any], path: str, default: Any | None = No
     return result
 
 
+def firewall_rule_id_from_payload(rule_key: Any, rule: Any) -> str | None:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `rule_key`

(ANN401)